exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusUnitsHelper.java

```diff
@@ -0,0 +1,225 @@
+      return "ratio";
+    }
+
+    // Drop units specified between curly braces
+    String convertedMetricUnitName = removeUnitPortionInBraces(rawMetricUnitName);
+    // Handling for the "per" unit(s), e.g. foo/bar -> foo_per_bar
+    convertedMetricUnitName = convertRateExpressedToPrometheusUnit(convertedMetricUnitName);
+    // Converting abbreviated unit names to full names
+    return cleanUpString(getPrometheusUnit(convertedMetricUnitName));
+  }
+
+  /**
+   * This method is used to convert the units expressed as a rate via '/' symbol in their name to
+   * their expanded text equivalent. For instance, km/h => km_per_hour. The method operates on the
+   * input by splitting it in 2 parts - before and after '/' symbol and will attempt to expand any
+   * known unit abbreviation in both parts. Unknown abbreviation will remain unchanged in the final
+   * output.
+   *
+   * @param rateExpressedUnit The rate unit input that needs to be converted to its text equivalent.
+   * @return The text equivalent of unit expressed as rate. If the input does not contain '/', the
+   *     function returns it as-is.
+   */
+  private static String convertRateExpressedToPrometheusUnit(String rateExpressedUnit) {
+    if (!rateExpressedUnit.contains("/")) {
+      return rateExpressedUnit;
+    }
+    String[] rateEntities = rateExpressedUnit.split("/", 2);
+    // Only convert rate expressed units if it's a valid expression
+    if (rateEntities[1].equals("")) {
+      return rateExpressedUnit;
+    }
+    return getPrometheusUnit(rateEntities[0]) + "_per_" + getPrometheusPerUnit(rateEntities[1]);
+  }
+
+  /**
+   * This method drops all characters enclosed within '{}' (including the curly braces) by replacing
+   * them with an empty string. Note that this method will not produce the intended effect if there
+   * are nested curly braces within the outer enclosure of '{}'.
+   *
+   * <p>For instance, {packet{s}s} => s}.
+   *
+   * @param unit The input unit from which text within curly braces needs to be removed.
+   * @return The resulting unit after removing the text within '{}'.
+   */
+  private static String removeUnitPortionInBraces(String unit) {
+    return unit.replaceAll(CHARACTERS_BETWEEN_BRACES_REGEX, "");
+  }
+
+  /**
+   * Replaces all characters that are not a letter or a digit with '_' to make the resulting string
+   * Prometheus compliant. This method also removes leading and trailing underscores.
+   *
+   * @param string The string input that needs to be made Prometheus compliant.
+   * @return the cleaned-up Prometheus compliant string.
+   */
+  private static String cleanUpString(String string) {
+    String prometheusCompliant =
+        INVALID_CHARACTERS_PATTERN.matcher(string).replaceAll("_").replaceAll("[_]{2,}", "_");
+    prometheusCompliant = prometheusCompliant.replaceAll("_+$", ""); // remove trailing underscore
+    prometheusCompliant = prometheusCompliant.replaceAll("^_+", ""); // remove leading underscore
```

Comment:
What's the reason for these replacements? Prometheus does not seem to forbid double or trailing underscores in its spec; and it does not seem that the previous operations on the unit are likely to produce trailing or duplicated underscores.

Reply:
The spec linked in the PR  description states - 
```
 Multiple consecutive `_` characters MUST be replaced with a single `_` character
```

Based on that I removed leading `_`, because the serializer was already adding it.  
```
name = name + "_" + prometheusEquivalentUnit;
```

But looking at this again, I think this would be confusing to do in the `PrometheusUnitHelper`.  I will remove this, also the spec does not explicitly mention that the final unit name cannot have a trailing `_`, so I think leading `_` is fine. I will make changes. 

Also, the current name sanitizer does not seem to remove consecutive `_`, so based on spec this needs to be added as well.

#### TL;DR
 - The PrometheusUnitHelper will just return the computed unit (with illegal characters replaced with `_`) without any additional cleaning up. 
 - The Sanitizing will be done by the serializer right before returning. The sanitizing will only take care of replacing any remaining Illegal characters with `_` and then replacing consecutive `_` with a single underscore.

Reply:
I would like to mention here that while not explicitly stated in the spec, this is how opentelemetry-collector [works](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/9a9d4778bbbf242dba233db28e2fbcfda3416959/pkg/translator/prometheus/normalize_name.go#L277), should we try to keep this in sync with how the collector works ?